spec/lucky/session_handler_spec.cr

```diff
@@ -0,0 +1,119 @@
+require "../spec_helper"
+
+include ContextHelper
+
+describe Lucky::SessionHandler do
+  it "sets a cookie" do
+    context = build_context
+    context.cookies.set(:email, "test@example.com")
+
+    Lucky::SessionHandler.new.call(context)
+
+    context.response.headers.has_key?("Set-Cookie").should be_true
+    context.response.headers["Set-Cookie"].should contain("email=")
+  end
+
+  it "persists the cookies across multiple requests" do
+    context_1 = build_context
+    context_1.cookies.set(:email, "test@example.com")
+    Lucky::SessionHandler.new.call(context_1)
+
+    request = build_request
+    cookie_header = context_1.response.cookies.map do |cookie|
+      cookie.to_cookie_header
+    end.join(", ")
+    request.headers.add("Cookie", cookie_header)
```

Comment:
Could you explain why cookies.set and setting the header is required? I’m not sure I understand

Reply:
I think this makes sense, but it isn't super clear from the code. Maybe adding something to the description like `persists the cookies across multiple requests when set with header or in Lucky`. That sounds horrible, but maybe that triggers some ideas?

Reply:
Good distinction of "proper code but description could be clearer" 👍 Updated the description and some variables in a61bc1b.